### ISSUE TITLE
SignalR Hub Example

### ISSUE THREAD

**Issue by gswartz777** [Cysharp/Claudia#20]
I'm trying to use this in a .net 6 web app with SignalR.  I'm running into a couple of problems.  Basically the problem is I'm not sure how to stream back the result as a string to the client.  I'm not sure if the following code should work or not.  Unfortunately I can't test it because I'm getting the error "Invalid expression term '['" on the Messages = [ line.

```
public async Task<string> Chat(AiChatRequestViewModel requestViewModel,
        [EnumeratorCancellation]
        CancellationToken cancellationToken)
    {
        var anthropic = new Anthropic();

        var stream = anthropic.Messages.CreateStreamAsync(new()
        {
            Model = "claude-3-opus-20240229",
            MaxTokens = 1024,
            Messages = [new Message() { Role = "user", Content = "Hello, Claude" }]
        }, cancellationToken: cancellationToken);
        
        await foreach (var messageStreamEvent in stream)
        {
            return messageStreamEvent.Type;
        }
    }
```

I'm also not sure if messageStreamEvent.Type is what I want to return or if I need to perform additional logic on that.

**Comment by neuecc** [comment]
stream is stream, you should return `IAsyncEnumerable<string>`.

**Comment by gswartz777** [comment]
That's for .net 8.  As I said, we're on .net 6.  Unfortunately we can't move to 8 for quite a while.

**Comment by gswartz777** [comment]
I'm assuming this is what you're thinking?  Still can't get it to compile for .net 6 even though it doesn't complain about the IAsyncEnumerable and yield.  I just need to see how to set up the Messages property for .net 6.

```
public async IAsyncEnumerable<string> Chat(AiChatRequestViewModel requestViewModel,
        [EnumeratorCancellation]
        CancellationToken cancellationToken)
    {
        var anthropic = new Anthropic()
        {
            ApiKey = AwsSecretsManager.AnthropicApiKey
        };

        var stream = anthropic.Messages.CreateStreamAsync(new()
        {
            Model = "claude-3-opus-20240229",
            MaxTokens = requestViewModel.Samples,
            Temperature = 0.0,
            Messages = [new () { Role = "user", Content = requestViewModel.AIPrompt }]
        }, cancellationToken: cancellationToken);
        
        await foreach (var messageStreamEvent in stream)
        {
            yield return messageStreamEvent.Type;
        }
    }
```

**Comment by neuecc** [comment]
What about Broadcasting to the Client?
What about making the Chat Request a continuous request to the client?

Note that this has become a SignalR question, so Close.

**Comment by gswartz777** [comment]
Thanks for suggesting things that don't actually relate to the question.  It's not a SignalR issue.  It's a question about how to get this to work with .net6.  The readme specifically says "We have built a C#... It supports netstandard2.1, **net6.0**, and net8.0."  

Well if it supports net 6, why won't it compile when referenced in a .net 6 app?  Honestly though, this is now irrelevant since I found the Anthropic.SDK which works perfectly to stream the results and it does work in .net 6.
